pkg/shp/cmd/build/run_test.go

```diff
@@ -0,0 +1,90 @@
+	shpfake "github.com/shipwright-io/build/pkg/client/clientset/versioned/fake"
+	"github.com/shipwright-io/cli/pkg/shp/flags"
+	"github.com/shipwright-io/cli/pkg/shp/params"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes/fake"
+	fakekubetesting "k8s.io/client-go/testing"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestStartBuildRunFollowLog(t *testing.T) {
+	//		"build.shipwright.io/name=%s,buildrun.shipwright.io/name=%s",
+	name := "testpod"
+	containerName := "container"
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: metav1.NamespaceDefault,
+			Name:      name,
+			Labels: map[string]string{
+				buildv1alpha1.LabelBuild:    name,
+				buildv1alpha1.LabelBuildRun: name,
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Name: containerName,
+			}},
+		},
+	}
+	br := &buildv1alpha1.BuildRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: metav1.NamespaceDefault,
+			Name:      name,
+		},
+	}
+	shpclientset := shpfake.NewSimpleClientset()
+	createReactorFunc := func(action fakekubetesting.Action) (handled bool, ret runtime.Object, err error) {
+		return true, br, nil
+	}
+	shpclientset.PrependReactor("create", "buildruns", createReactorFunc)
+	kclientset := fake.NewSimpleClientset(pod)
+	ccmd := &cobra.Command{}
+	cmd := RunCommand{
+		cmd:             ccmd,
+		buildRunName:    name,
+		buildRunSpec:    flags.BuildRunSpecFromFlags(ccmd.Flags()),
+		follow:          true,
+		shpClientset:    shpclientset,
+		tailLogsStarted: make(map[string]bool),
+	}
+
+	// set up context
+	cmd.Cmd().ExecuteC()
+	param := params.NewParamsForTest(kclientset, shpclientset, nil, metav1.NamespaceDefault)
+
+	ioStreams, _, out, _ := genericclioptions.NewTestIOStreams()
+
+	cmd.Complete(param, []string{name})
+	go func() {
+		err := cmd.Run(param, &ioStreams)
+		if err != nil {
+			t.Fatalf("%s", err.Error())
+		}
+
+	}()
+
+	// after much back and forth, going with a simple have we gotten far enough along the cmd.Run path;
+	// employing simple sleep; use of Run variables requires locking or we get DATA RACE with our unit tests with race detection;
+	// Once we are far enough along, we then call the event with the completed pod
+	// which should cause the pod watcher to exit
+	// graceful time to wait for container start
+	time.Sleep(3 * time.Second)
+	// mimic watch eventa, bypassing k8s fake client watch hoopla whose plug points are not always useful;
+	// we are the logic we supply into this package to other packages like tail and pod watcher;
+	// they each have unit tests that test their logic
+	pod.Status.Phase = corev1.PodSucceeded
+	cmd.onEvent(pod)
+	if !strings.Contains(out.String(), "Pod 'testpod' has succeeded!") {
+		t.Fatalf("unexpected output: %s", out.String())
```

Comment:
nit: alternatively we could use `gomega` assertion library, as we do in the other unit-tests.

Reply:
yeah I've see that and am not the biggest fan ... was hoping for some developer choice here :-) 

but let me ponder and then "officially" respond